rqt_py_common/src/rqt_py_common/topic_helpers.py

```diff
@@ -46,130 +50,259 @@ def get_topic_names_and_types(node=None):
-    #       In ROS2 multiple msg types can be used with a single topic making this
-    #       funciton a bad candidate to port to ROS2
-    logger = logging.get_logger("topic_helpers")
-    logger.error("get_field_type is not implemented in ROS2")
-    # get topic_type and message_evaluator
-    # topic_type, real_topic_name, _ = get_topic_type(topic_name)
-    # if topic_type is None:
-    #     # qDebug('topic_helpers.get_field_type(%s): get_topic_type failed' % (topic_name))
-    #     return None, False
-
-    # message_class = roslib.message.get_message_class(topic_type)
-    # if message_class is None:
-    #     qDebug('topic_helpers.get_field_type(%s): get_message_class(%s) failed' %
-    #            (topic_name, topic_type))
-    #     return None, False
-
-    # slot_path = topic_name[len(real_topic_name):]
-    # return get_slot_type(message_class, slot_path)
-    pass
+    topic_names_and_types = get_topic_names_and_types(node=node)
+    return _get_field_type(topic_names_and_types, target)
+
+
+
+def _get_field_type(topic_names_and_types, target):  # noqa
+    """
+    Testable helper function for get_field_type.
+    """
+    logger = logging.get_logger('topic_helpers._get_field_type')
+    for name, types in topic_names_and_types:
+        # If the topic is a substring of the target param, then we have found a match
+        if target.find(name) >= 0:
+            # If the target passed in was the topic address not the address of a field
+            if target == name:
+                # If there is more than one type of topic on target
+                if len(types) > 1:
+                    logger.warn(
+                        'Ambiguous request. Multiple topic types found on: {}'.format(target))
+                    return None, False
+                # If the types array is empty then something weird has happend
+                if len(types) == 0:
+                    logger.warn(
+                        'Ambiguous request. No topic types found on: {}'.format(target))
+                    return None, False
+
+                # If there is only one msg class on
+                msg_type_str = types[0]
+                msg_class = get_type_class(msg_type_str)
+                return msg_class, False
+
+            else:
+                # The topic must be a substring of the target
+
+                # Get the address of the field in the messgage class
+                field_address_str = target[len(name):]
+
+                # Iterate through the message types on the given topic and see if any match the
+                # path that was provided
+                for msg_type_str in types:
+                    try:
+                        msg_class = get_message_class(msg_type_str)
+
+                        # If the message type is a simple type eg: float, then it can't be an array
+                        if _is_primative_type(msg_type_str):
+                            return msg_class, False
```

Comment:
```get_field_type``` is supposed to return the Python type of a field, not the full message class. For example, /chatter/_data resolves to std_msgs.msg._string.String

Reply:
Addressed in #155